sdk/ai/Azure.AI.Agents.Persistent/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -1222,25 +1227,63 @@ public static IReadOnlyDictionary<string, string> GetToolCallAttributes(RunStepT
                 throw new ArgumentNullException(nameof(toolCall));
             }
 
-            // Get the actual type of the toolCall instance
-            Type actualType = toolCall.GetType();
```

Comment:
The original implementation used non-AOT-safe reflection to find public property with name "yyy" where "yyy" is `RunStep{yyy}ToolCall` and are of type `Dictionary<string, string>`. The updated code eliminates reflection and implements the same behavior using explicit pattern matching."